### PR TITLE
Add Github Action to update kopia image and raise PR to update Kanister tools Dockerfile

### PR DESCRIPTION
**Pull request by ankitjain235**
## Change Overview

This PR adds a Github Action that can be manually triggered to build Kopia image using the commit id and raise a PR to update the kanister-tools image to use the respective image.

## Pull request type

Please check the type of change your PR introduces:
- [x] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: .github/workflows/update-kopia-image.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - uses: actions/checkout@v2
+    - run: make update-kopia-image COMMIT_ID=${{ github.event.inputs.commitId }}
+  updateref:
+    runs-on: ubuntu-20.04
+    needs: [build]
+    steps:
+    - uses: actions/checkout@v2
+    - run: make update-image-reference KOPIA_IMAGE_TAG="ghcr.io/kanisterio/kopia:alpine-${{ github.event.inputs.commitId }}"
+    - uses: peter-evans/create-pull-request@v4
```
Comment (pavannd1) [inline .github/workflows/update-kopia-image.yaml:29]: I know this is the only one with a decent amount of stars in the marketplace but we should make sure it's safe to use.

File: .github/workflows/update-kopia-image.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        branch-suffix: random
+        base: master
+        add-paths: |
+          docker/tools/Dockerfile
+        commit-message: "Update kanister tools image to use alpine-${{ github.event.inputs.commitId }} tag"
+        author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+        signoff: true
+        title: 'Update kanister tools image to use alpine-${{ github.event.inputs.commitId }} tag'
+        labels: test
+        draft: false
```
Comment (pavannd1) [inline .github/workflows/update-kopia-image.yaml:41]:
```suggestion
        draft: false

```

File: Makefile [hunk 3]
```diff
@@ -264,6 +266,9 @@ release-snapshot:
 update-kopia-image:
 	@/bin/bash ./build/update_kopia_image.sh $(COMMIT_ID) $(KOPIA_REPO)
 
+update-image-reference:
```
Comment (pavannd1) [inline Makefile:269]: When we improve the kopia image update to be a manual workflow, this should change as well. A successful run of the image update workflow should trigger this.

**Review by ihcsim — COMMENTED** [review]
Does this need to be a separate manual workflow? Can this be folded into the `main` workflow, with the Kopia version recorded in `.goreleaser.yaml`, making it the source of truth of all dependencies version? I think it's safe to say we would want to do a new release, every time we update Kopia, right?

File: .github/workflows/update-kopia-image.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        branch: kopia-dependency-upgrade/${{ github.event.inputs.commitId }}
+        branch-suffix: random
+        base: master
+        add-paths: |
+          docker/tools/Dockerfile
+        commit-message: "Update kanister tools image to use alpine-${{ github.event.inputs.commitId }} tag"
+        author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+        signoff: true
+        title: 'Update kanister tools image to use alpine-${{ github.event.inputs.commitId }} tag'
+        labels: test
```
[the remark this answers is not in the document]
Comment (ankitjain235) [reply .github/workflows/update-kopia-image.yaml:40]: No, it was just temporary, to be replaced by `kueue` label.

**Comment by ihcsim** [comment]
@pavannd1 @ankitjain235 Alternately, in light of the ongoing Kopia integration work where we will need to spin up Kopia servers, can we just keep the Kopia image, but remove the `kopia` CLI from the `kanister-tools` image? If folks want to run Kopia in K8s, just use the Kopia image. That should simplify the pipeline to just building the Kopia image as part of `.goreleaser.yaml`.

**Comment by ankitjain235** [comment]
Closing wrt https://github.com/kanisterio/kanister/pull/1711

**Pull request closed without merging**